amundsen_application/static/js/components/SearchPage/SearchFilter/CheckBoxFilter/index.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import { clearFilterByCategory, updateFilterByCategory } from 'ducks/search/filters/reducer';
+
+import CheckBoxItem from 'components/common/Inputs/CheckBoxItem';
+
+export interface CheckboxFilterProperties {
+  checked: boolean;
+  labelText: string;
+  value: string;
+}
+
+interface OwnProps {
+  categoryId: string;
+  properties: CheckboxFilterProperties[];
+}
+
+interface DispatchFromProps {
+  onCheckboxChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export type CheckBoxFilterProps = OwnProps & DispatchFromProps;
+
+export class CheckBoxFilter extends React.Component<CheckBoxFilterProps> {
+  constructor(props) {
+    super(props);
+  }
+
+  createCheckBoxItem = (categoryId: string, key: string, item: CheckboxFilterProperties) => {
+    const { checked, labelText, value } = item;
+    return (
+      <CheckBoxItem
+        key={key}
+        checked={ checked || false }
+        name={ categoryId }
+        value={ value }
+        onChange={ this.props.onCheckboxChange }>
+          <span className="subtitle-2">{ labelText }</span>
+      </CheckBoxItem>
+    );
+  };
+
+  render = () => {
+    const { categoryId, properties } = this.props;
+    return (
+      <div className="checkbox-section-content">
+        { properties.map((item, index) => this.createCheckBoxItem(categoryId, `item:${categoryId}:${index}`, item)) }
+      </div>
+    )
+  }
+};
+
+export const mapDispatchToProps = (dispatch: any, ownProps: OwnProps) => {
+  return {
+    onCheckboxChange: (e: React.ChangeEvent<HTMLInputElement>) => {
+      const checkedValues = {};
+      const value = e.target.value;
+      const category = e.target.name;
+      let shouldClearFilter = true;
+
+      ownProps.properties.forEach((property) => {
+        if ((property.value === value && e.target.checked) || (property.value !== value && property.checked)) {
+          checkedValues[property.value] = true;
+          shouldClearFilter = false;
+        }
+      });
+
+      if (shouldClearFilter) {
+        dispatch(clearFilterByCategory(category));
+      }
+      else {
+        dispatch(updateFilterByCategory(category, checkedValues));
```

Comment:
This is because I've implemented this work that does not include the category in the list of filters if it has no options. So in the reducer `updateFilterByCategory` is always updating the value of a particular category key, and `clearFilterByCategory` will remove that category key from the dictionary. 

If I wanted to use the same action, then the reducer logic would have to have some boolean logic. I went with having a separate action as opposed to having branching logic in the reducer.

FYI One could ask why we can't just have the category still in the filters but with an empty list or empty string as a value, but in a world where we have more categories and more potential values -- as users select and deselect -- I want to avoid bloating our application state with something like:
```
filters = {
   'database': { 'option1': false, ...all false options..., 'option20': false },
   'column': '',
   'table': '',
}
```
because all of that is equivalent to:
```
filters = {}
```